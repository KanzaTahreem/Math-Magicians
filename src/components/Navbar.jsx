import React from 'react';
import { NavLink } from 'react-router-dom';
import '../stylesheets/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="title">Math Magicians</h1>
    <ul className="links">
      <li className="linkItem">
        <NavLink
          to="/"
          end
          className={({ isActive }) => (isActive ? 'activelink' : undefined)}
        >
          Home
        </NavLink>
      </li>
      <li className="divider" />
      <li className="linkItem">
        <NavLink
          to="calculator"
          className={({ isActive }) => (isActive ? 'activelink' : undefined)}
        >
          Calculator
        </NavLink>
      </li>
      <li className="divider" />
      <li className="linkItem">
        <NavLink
          to="quote"
          className={({ isActive }) => (isActive ? 'activelink' : undefined)}
        >
          Quote
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
