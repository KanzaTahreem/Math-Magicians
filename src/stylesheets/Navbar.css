.activelink {
  text-decoration: underline;
  color: red;
}
