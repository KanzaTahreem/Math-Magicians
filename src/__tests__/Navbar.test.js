import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

it('renders correctly', () => {
  const navbar = renderer
    .create(<Router><Navbar /></Router>)
    .toJSON();
  expect(navbar).toMatchSnapshot();
});
