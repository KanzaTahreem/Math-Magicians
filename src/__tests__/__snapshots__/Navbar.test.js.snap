// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar"
>
  <h1
    className="title"
  >
    Math Magicians
  </h1>
  <ul
    className="links"
  >
    <li
      className="linkItem"
    >
      <a
        aria-current="page"
        className="activelink"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="divider"
    />
    <li
      className="linkItem"
    >
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="divider"
    />
    <li
      className="linkItem"
    >
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
